src/aktualizr_primary/main.cc

```diff
@@ -122,11 +122,14 @@ int main(int argc, char *argv[]) {
     conn = aktualizr.SetSignalHandler(f_cb);
 
     if (!config.uptane.secondary_config_file.empty()) {
-      if (boost::filesystem::exists(config.uptane.secondary_config_file)) {
+      try {
```

Comment:
This try/catch was not necessary as this code is already wrapped by the outter try/catch.

Reply:
The exception is ignored if `aktualizr.IsRegistered()`, I think we cannot rely on the outer try/catch in this case.

Reply:
ok. I see, this is for the use case when Primary and Secondaries are trying to connect to each other after a device has been already provisioned, e.g. just a device and/or aktualizr reboot. I think, it might be not an optimal strategy to exit the connection procedure on a first failure if a device has been already registered, but, I suppose, it's not so important, adding secondaries dynamically is more important or not trying to connect to them on a startup at all if  a device already registered.

Reply:
> I think, it might be not an optimal strategy to exit the connection procedure on a first failure if a device has been already registered, but, I suppose, it's not so important, adding secondaries dynamically is more important or not trying to connect to them on a startup at all if a device already registered.

That's a good point indeed... But I remember I chose this simple solution at first because right now, we can't really do better than what the IP secondaries config parser gives us and it handles all IP secondaries at once.